src/fhirtypes/common.ts

```diff
@@ -1255,6 +1294,42 @@ export function getUrlFromFshDefinition(
   return `${canonical}/${fhirType}/${fshDefinition.id}`;
 }
 
+/**
+ * Determines the version of this entity.
+ * If a caret value rule has been applied to the entity's version, use the value specified in that
+ * rule. Otherwise, use the default version based on the configured version from the tank.
+ *
+ * @param fshDefinition - The FSH definition whose version is being determined
+ * @param canonical - The version for the FSH project
+ * @returns The version of the FHIR entity
+ */
+export function getVersionFromFshDefinition(
+  fshDefinition:
+    | Profile
+    | Extension
+    | Logical
+    | Resource
+    | FshValueSet
+    | FshCodeSystem
+    | Instance
+    | Invariant
+    | RuleSet
+    | Mapping,
+  version: string
+): string {
+  // TODO - this early exit can be removed once rules are supported on invariants
+  if (fshDefinition instanceof Invariant) {
+    return version;
+  }
```

Comment:
An `Invariant` will never have a version in it. Nor will a `Mapping`.  I guess a `RuleSet` might, but I'm not sure that's relevant here since you don't know where the RuleSet is applied?